### PR TITLE
feat(datasets): Add kwargs for huggingface.HFDataset

### PR DESCRIPTION
**Pull request by eromerobilbomatica**
## Description

Issue: The kedro_datasets.huggingface.HFDataset class have to accept parameters optionals in call of the function #574

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by datajoely** [comment]
thanks for raising this @eromerobilbomatica !

**Comment by noklam** [comment]
@eromerobilbomatica I see a lot of newlines get added in files, is it possible to not include these changes in the PR? Ideally this PR should touch only `kedor-datasets` but not other plugins. 

p.s. I also updated the title

**Comment by eromerobilbomatica** [comment]
That is @noklam, my PR only is connected `kedor-datasets`. The rest of the chances have been created when I run `make plugin=kedro-datasets lint`. Thanks for fixing the title.

**Comment by astrojuanlu** [comment]
@eromerobilbomatica There is a legitimate problem, see the failed checks:

```
kedro.io.core.DatasetError: Failed while loading data from data set HFDataset(dataset_name=openai_humaneval, dataset_tags=['task_categories:text2text-generation', 'annotations_creators:expert-generated', 'language_creators:expert-generated', 'multilinguality:monolingual', 'size_categories:n<1K', 'source_datasets:original', 'language:en', 'license:mit', 'code-generation', 'croissant', 'arxiv:2107.03374', 'region:us']).
'HFDataset' object has no attribute '_pipeline_kwargs'
```

**Comment by eromerobilbomatica** [comment]
@astrojuanlu just fixed it, sorry! things of copy and paste.

**Comment by astrojuanlu** [comment]
Hi @eromerobilbomatica , this is mostly ready but the DCO check doesn't pass and rebasing this is tricky... Can you either squash everything together or open a new pull request?

**Review by merelcht — APPROVED** [review]
Thanks for your contribution @eromerobilbomatica ! ⭐  Can you add the change to the release notes together with your name as contributor (you can see examples in old release notes).

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -44,13 +44,15 @@ Many thanks to the following Kedroids for contributing PRs to this release:
...
 * Correct pandas-gbq as py311 dependency
+* Fixed bug with loading datasets from Hugging Face.
 
 ## Community contributions
 Many thanks to the following Kedroids for contributing PRs to this release:
 * [Edouard59](https://github.com/Edouard59)
 * [Miguel Rodriguez Gutierrez](https://github.com/MigQ2)
 * [felixscherz](https://github.com/felixscherz)
 * [Onur Kuru](https://github.com/kuruonur1)
+* [Eduardo Romero Lopez](https://github.com/eromerobilbomatica)
```
Comment (ankatiyar) [inline kedro-datasets/RELEASE.md:55]: @eromerobilbomatica These changes should be under the "Upcoming Release" heading at the top of this page!

**Comment by astrojuanlu** [comment]
Woah, it's trying to install Dask 0.8.2, released 2016... https://pypi.org/project/dask/0.8.2/ something very weird going on

**Comment by merelcht** [comment]
> Woah, it's trying to install Dask 0.8.2, released 2016... https://pypi.org/project/dask/0.8.2/ something very weird going on

Dask just released a new version: https://pypi.org/project/dask/2024.3.0/#history, pinning it seems to at least fix the installation step but there's other stuff wrong as well: https://github.com/kedro-org/kedro-plugins/pull/604

**Review by noklam — APPROVED** [review]

**Comment by astrojuanlu** [comment]
Never ending story

**Comment by astrojuanlu** [comment]
@eromerobilbomatica now the `yaml.org` site disappeared and our links broke, we're waiting a few hours to see if the problem fixes itself

**Pull request merged**